### PR TITLE
Verify/Correct input values and go to next input automaticaly

### LINKED ISSUE #128: Writing dates without jumping between date/month/year fields
Hello!
I was asked if it's possible to be "faster" in writing a date.
Right now, if I want to use the keyboard to write the date, if I for example want to insert the date 12/03/2019, I need to type: 12 *tab* 03 *tab* 2019... but some users of some other environment are really use to do it faster, like writing 12032019. This because usually they have an automatic focus after 2 digits on the first input and after other 2 digits on the second input...
I would like to do the same thing, but obviously the component doesn't let me, I didn't find a way to have an event after every digit I press...

Can I do it some other way?

thank you.

### PR DESCRIPTION
**Pull request by NicoDos**
Fixes #128

I want to improve ux by automatically focus next input when the current one is filled correctly.
This is pretty much the same behaviour as the html5 date input (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date). Additionally it prevents wrong input values.
Looking forward to your feedback.

### CODE REVIEW

**Comment by NicoDos** [comment]
Hi @wojtekmaj,
Some time to review this PR?

thanks!

File: .gitignore [hunk 1]
```diff
@@ -2,7 +2,6 @@
 .idea
 .vscode
 .idea
-dist
```
Comment (wojtekmaj) [inline .gitignore]: Please don't commit `dist` folder to the repository. Automatically generated code should not be commited. Just like `node_modules` :)

File: src/DateInput/DayInput.jsx [hunk 2]
```diff
@@ -64,8 +64,8 @@ export default class DayInput extends PureComponent {
         onChange={onChange}
         onFocus={event => select(event.target)}
         onKeyDown={onKeyDown}
-        onKeyUp={event => updateInputWidth(event.target)}
-        placeholder="--"
+        onKeyUp={event => appendInputValue(event, onChange)}
```
Comment (wojtekmaj) [inline src/DateInput/DayInput.jsx:67]: Hmm, so now this function does not run on key up? I think this improved input behavior 🤔

File: src/DateInput/MonthInput.jsx [hunk 3]
```diff
@@ -47,8 +47,8 @@ export default class MonthInput extends PureComponent {
         onChange={onChange}
         onFocus={event => select(event.target)}
         onKeyDown={onKeyDown}
-        onKeyUp={event => updateInputWidth(event.target)}
-        placeholder="--"
+        onKeyUp={event => appendInputValue(event, onChange)}
+        placeholder="mm"
```
Comment (wojtekmaj) [inline src/DateInput/MonthInput.jsx]: Let's focus on one change at a time. Changing placeholders is discussed in another issue.

File: src/DateInput/__tests__/DayInput.jsx [hunk 4]
```diff
@@ -237,4 +237,28 @@ describe('DayInput', () => {
...
+        year={2018}
+        month={1}
+        maxDate={new Date(2018, 0, 15)}
+      />
+    );
+
+    component.simulate('keyUp', event);
+
+    expect(onChange).toHaveBeenCalled();
+    expect(onChange).toHaveBeenCalledWith(expect.objectContaining(event));
```
Comment (wojtekmaj) [inline src/DateInput/__tests__/DayInput.jsx:262]: Nice, I didn't know this :)

File: src/shared/__tests__/utils.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+    it('triggers onChange correctly', () => {
+      expect(onChange).toHaveBeenCalled();
+      expect(onChange).toHaveBeenCalledWith(event);
+    });
+  });
+});
+
+describe('isPositiveInteger', () => {
+  it('should not accept e', () => {
```
Comment (wojtekmaj) [inline src/shared/__tests__/utils.js:38]: Sounds like you need it.each :)

**Review by wojtekmaj — CHANGES_REQUESTED** [review]
Nicely done and seems to be well tested. I don't understand everything just yet, but I think this may be a way to go after some adjustments :)

File: .gitignore (hunk 1 above)
Comment (NicoDos) [reply .gitignore]: Sorry this was not meant to be committed.

File: src/shared/__tests__/utils.js (hunk 5 above)
Comment (NicoDos) [reply src/shared/__tests__/utils.js:38]: Not sure I understand, do you prefer only one it() with all cases? :)

File: src/DateInput/MonthInput.jsx (hunk 3 above)
Comment (NicoDos) [reply src/DateInput/MonthInput.jsx]: Agreed and reverted.

File: src/DateInput/__tests__/DayInput.jsx (hunk 4 above)
Comment (NicoDos) [reply src/DateInput/__tests__/DayInput.jsx:262]: Thanks!

**Review by NicoDos — COMMENTED** [review]
Hi, thank you for reviewing.

**Comment by NicoDos** [comment]
> Nicely done and seems to be well tested. I don't understand everything just yet, but I think this may be a way to go after some adjustments :)

Thank you, I think I am done.

**Comment by nnaku** [comment]
Is this getting merged sooner or later?

**Comment by varunthefalcon** [comment]
Does this feature come with the package or should we pass any props(if so needed documentation). 
I couldn't see this implementation. 

P.S : All packages upto date

**Comment by wojtekmaj** [comment]
@varunthefalcon In the meantime a major rewrite was coming so I'm working on re-implementing this based on @NicoDos's work. Will be available soon.

**Pull request closed without merging**